pkg/probe/http/http.go

```diff
@@ -100,19 +100,19 @@ func DoHTTPProbe(req *http.Request, client GetHTTPInterface) (probe.Result, stri
 	}
 	defer res.Body.Close()
 	b, err := utilio.ReadAtMost(res.Body, maxRespBodyLength)
-	if err != nil {
-		if err == utilio.ErrLimitReached {
-			klog.V(4).Infof("Non fatal body truncation for %s, Response: %v", url.String(), *res)
-		} else {
-			return probe.Failure, "", err
-		}
+	if err != nil && !errors.Is(err, utilio.ErrLimitReached) {
+		return probe.Failure, "", err
 	}
 	body := string(b)
 	if res.StatusCode >= http.StatusOK && res.StatusCode < http.StatusBadRequest {
 		if res.StatusCode >= http.StatusMultipleChoices { // Redirect
 			klog.V(4).Infof("Probe terminated redirects for %s, Response: %v", url.String(), *res)
 			return probe.Warning, fmt.Sprintf("Probe terminated redirects, Response body: %v", body), nil
 		}
+		if errors.Is(err, utilio.ErrLimitReached) {
+			klog.V(4).Infof("Non fatal body truncation for %s, Response: %v", url.String(), *res)
+			return probe.Warning, fmt.Sprintf("Non fatal body truncation for %s", url.String()), nil
```

Comment:
This looks like it will spawn a warning event API object to be created. Do we know how often that will happen?